airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py

```diff
@@ -83,7 +108,7 @@ def list_files(self) -> List[RemoteFile]:
         ...
 
     @abstractmethod
-    def list_files_for_this_sync(self, stream_slice: Optional[StreamSlice]) -> Iterable[RemoteFile]:
+    def list_files_for_this_sync(self) -> Iterable[RemoteFile]:
```

Comment:
I find this interface a little weird. It implies the notion of sync, but it's not passed anywhere. The reason seems to be that the object has a state and can't be reused for more than one sync. Do we need to make this object stateful?